### PR TITLE
Allow for configuring etcd servers addresses in gce

### PR DESCRIPTION
**Pull request by wojtek-t**

### CODE REVIEW

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -1461,8 +1461,15 @@ function start-kube-apiserver {
   params+=" --allow-privileged=true"
   params+=" --cloud-provider=gce"
   params+=" --client-ca-file=${CA_CERT_BUNDLE_PATH}"
-  params+=" --etcd-servers=http://127.0.0.1:2379"
-  params+=" --etcd-servers-overrides=/events#http://127.0.0.1:4002"
+  if [[ -n "${ETCD_SERVERS:-}" ]]; then
+    params+=" --etcd-servers=${ETCD_SERVERS:-}"
+    if [[ -n "${ETCD_SERVERS_OVERRIDES:-}" ]]; then
+      params+=" --etcd-servers-overrides=${ETCD_SERVERS_OVERRIDES:-}"
```
Comment (shyamjvs) [inline cluster/gce/gci/configure-helper.sh]: Seems like you can remove this if condition and just move similar line in the else block below outside.

Comment (wojtek-t) [reply cluster/gce/gci/configure-helper.sh]: No - these are two different things.

Here (if you explicitly specifi etcd-server), you also need to expicitly specify overrides (if not your events would be send to the same etcd).
In the "else" block, we default to local etcd, but you may override it.

You can't achieve that in a single instruction.

Comment (shyamjvs) [reply cluster/gce/gci/configure-helper.sh]: Why can't you just do the following?

```
params+=" --etcd-servers=${ETCD_SERVERS:-http://127.0.0.1:2379}"
if [[ -z "${ETCD_SERVERS:-}" ]]; then
  ETCD_SERVERS_OVERRIDES=${ETCD_SERVERS_OVERRIDES:-/events#http://127.0.0.1:4002}
fi
params+=" --etcd-servers-overrides=${ETCD_SERVERS_OVERRIDES:-}"
```

Comment (wojtek-t) [reply cluster/gce/gci/configure-helper.sh]: Setting the flag to empty value and not setting it aren't the same things IIRC.
Also we don't want to override variable here, I think.

I moved the etcd-servers= outside ifs.

Comment (shyamjvs) [reply cluster/gce/gci/configure-helper.sh]: If that's the case, you still don't need a nested if-else. You can:

```
params+=" --etcd-servers=${ETCD_SERVERS:-http://127.0.0.1:2379}"
if [[ -z "${ETCD_SERVERS:-}" ]]; then
  ETCD_SERVERS_OVERRIDES=${ETCD_SERVERS_OVERRIDES:-/events#http://127.0.0.1:4002}
fi
if [[ -n "${ETCD_SERVERS_OVERRIDES:-}" ]]; then
  params+=" --etcd-servers-overrides=${ETCD_SERVERS_OVERRIDES:-}"
fi
```

Comment (wojtek-t) [reply cluster/gce/gci/configure-helper.sh]: As I wrote above:
`Also we don't want to override variable here, I think`

Comment (shyamjvs) [reply cluster/gce/gci/configure-helper.sh]: > Also we don't want to override variable here, I think

Sorry, why?

Comment (wojtek-t) [reply cluster/gce/gci/configure-helper.sh]: In my opinion it's very misleading when we are changing values of configuration env vars in the provisioning script. Some other parts of the script may in theory rely on that.

Comment (shyamjvs) [reply cluster/gce/gci/configure-helper.sh]: Yeah, I agree with you on that. Maybe then we should just do:

```
params+=" --etcd-servers=${ETCD_SERVERS:-http://127.0.0.1:2379}"
if [[ -n "${ETCD_SERVERS_OVERRIDES:-}" ]]; then
  params+=" --etcd-servers-overrides=${ETCD_SERVERS_OVERRIDES:-}"
fi
```

and set the default events override in config-default.sh?

Comment (wojtek-t) [reply cluster/gce/gci/configure-helper.sh]: That would work, but I think less spreaded that is, the better.
I don't see the problem with having additional nested if here.

Comment (shyamjvs) [reply cluster/gce/gci/configure-helper.sh]: (Sorry for nitpicking on this)

But if we're not doing that, then reason you gave above (of having config in code) still holds. Then I don't see how nested if is better than my suggestion (which basically is just a negated of the current if in order to get rid of nesting).

Comment (wojtek-t) [reply cluster/gce/gci/configure-helper.sh]: Discussed offline. Now it should much both our requirements.

PTAL

**Pull request merged**